lib/dor/services/workflow_service.rb

```diff
@@ -7,539 +7,450 @@
+    # @param [Hash] opts optional values for the workflow step
+    # @option opts [Float] :elapsed The number of seconds it took to complete this step. Can have a decimal.  Is set to 0 if not passed in.
+    # @option opts [String] :lifecycle Bookeeping label for this particular workflow step.  Examples are: 'registered', 'shelved'
+    # @option opts [String] :note Any kind of string annotation that you want to attach to the workflow
+    # @option opts [String] :lane_id Id of processing lane used by the job manager.  Can convey priority or name of an applicaiton specific processing lane (e.g. 'high', 'critical', 'hydrus')
+    # @option opts [String] :current_status Setting this string tells the workflow service to compare the current status to this value.  If the current value does not match this value, the update is not performed
+    # @return []
+    def update_workflow_status(repo, druid, workflow, process, status, opts = {})
+      raise ArgumentError, "Unknown status value #{status}" unless @valid_statuses.include?(status.downcase)
+      opts = { :elapsed => 0, :lifecycle => nil, :note => nil }.merge!(opts)
+      opts[:elapsed] = opts[:elapsed].to_s
+      current_status = opts.delete(:current_status)
+      xml = create_process_xml({ :name => process, :status => status.downcase }.merge!(opts))
+      uri = "#{repo}/objects/#{druid}/workflows/#{workflow}/#{process}"
+      uri << "?current-status=#{current_status.downcase}" if current_status
+      workflow_resource_method(uri, 'put', xml, { :content_type => 'application/xml' })
+    end
 
-      # Get workflow names into an array for given PID
-      # This method only works when this gem is used in a project that is configured to connect to DOR
-      #
-      # @param [String] pid of druid
-      # @param [String] repo repository for the object
-      # @return [Array<String>] list of worklows
-      # @example
-      #   Dor::WorkflowService.get_workflows('druid:sr100hp0609')
-      #   => ["accessionWF", "assemblyWF", "disseminationWF"]
-      def get_workflows(pid, repo = 'dor')
-        xml_doc = Nokogiri::XML(get_workflow_xml(repo, pid, ''))
-        xml_doc.xpath('//workflow').collect {|workflow| workflow['id']}
-      end
+    # Retrieves the process status of the given workflow for the given object identifier
+    # @param [String] repo The repository the object resides in.  Currently recoginzes "dor" and "sdr".
+    # @param [String] druid The id of the object
+    # @param [String] workflow The name of the workflow
+    # @param [String] process The name of the process step
+    # @return [String] status for repo-workflow-process-druid
+    def get_workflow_status(repo, druid, workflow, process)
+      doc = get_workflow_ngxml(repo, druid, workflow)
+      status = doc.root.at_xpath("//process[@name='#{process}']/@status")
+      status = status.content if status
+      status
+    end
```

Comment:
This is hard to compare with the prior method, only because it has moved location in the file.  In general, unless there are compelling reasons to group similar method, an alphabetical order of methods is easier to follow.  The exception is often to group the get/set methods that share the same suffix, otherwise the methods are ordered alphabetically by the suffix.  What is the significance of the `update` prefix vs. a `set` prefix to match the `get` prefixes?  I tend to equate a PUT with `set` and a PATCH with `update`.